misc/package/Data Files/MWSE/core/mcm/components/settings/OnOffButton.lua

```diff
@@ -10,13 +10,13 @@
 
 local Parent = require("mcm.components.settings.Button")
 
+
+
 --- @class mwseMCMOnOffButton
 local OnOffButton = Parent:new()
-OnOffButton.defaultSetting = false
 
-function OnOffButton:getText()
-	local text = (self.variable.value and tes3.findGMST(tes3.gmst.sOn).value or tes3.findGMST(tes3.gmst.sOff).value)
-	return text --[[@as string]]
+function OnOffButton:convertToLabelValue(variableValue)
+	return tes3.findGMST(variableValue and tes3.gmst.sOn or tes3.gmst.sOff).value
 end
```

Comment:
As I noted in YesNoButton, the base Component class stores `sYes` and `sNo`. Could you also add the `sOn` and `sOff` to the base Component class and use these values as fields here:
`return variableValue and self.sYes or self.sNo`

Reply:
oh that's really nice to know. i'll change both this and the `YesNoButton`

Reply:
Please add the `sYes` and `sNo` fields to the base Component class.